src/hierarchies/lapnig_hierarchy.h

```diff
@@ -0,0 +1,119 @@
+#ifndef BAYESMIX_LAPNIG_HIERARCHY_H
+#define BAYESMIX_LAPNIG_HIERARCHY_H
+
+#include <google/protobuf/stubs/casts.h>
+
+#include <Eigen/Dense>
+#include <memory>
+#include <set>
+#include <vector>
+
+#include "algorithm_state.pb.h"
+#include "base_hierarchy.h"
+#include "hierarchy_id.pb.h"
+#include "hierarchy_prior.pb.h"
+
+namespace LapNIG {
+//! Custom container for State values
+struct State {
+  double mean, scale;
+};
+
+//! Custom container for Hyperparameters values
+struct Hyperparams {
+  double mean, var, shape, scale, mh_log_scale_var, mh_mean_var;
+};
+}  // namespace LapNIG
+
+class LapNIGHierarchy
+    : public BaseHierarchy<LapNIGHierarchy, LapNIG::State, LapNIG::Hyperparams,
+                           bayesmix::LapNIGPrior> {
+ public:
+  LapNIGHierarchy() = default;
+  ~LapNIGHierarchy() = default;
+
+  //! Counters for tracking acceptance rate in MH step
+  static unsigned int accepted_;
+  static unsigned int iter_;
```

Comment:
Do these need to be static? This way, I think, you're counting all iterations across all hiearchies; is this intended? It seems weird

Reply:
We wanted to compute those for all hierarchies and not for each separately. Otherwise we would end up with a big list of acceptance rates which we thought would be difficult to interpret. Also, all the chains share the same variance parameter so we didn't find it necessary to look at them individually. If it's weird or irrelevant in your opinion, we can change it.

Reply:
I think it's fine to leave it like this at least for the moment (big changes ahead in the library in the next future anyways!)